app/models/piece.rb

```diff
@@ -3,6 +3,20 @@ class Piece < ActiveRecord::Base
 
   enum color: [:white, :black]
 
+  def move_to!(new_x, new_y)
+    piece = Game.find(game_id).pieces.find_by(x_coordinate: new_x, y_coordinate: new_y)
```

Comment:
since we have `belongs_to :game` in this model. You can take advantage of that and write this like...

`game.pieces.find_by(x_coordinate: new_x, y_coordinate: new_y)`